spec/models/performance_statistics_spec.rb

```diff
@@ -31,8 +34,39 @@
 
     stats = PerformanceStatistics.new
 
-    expect(stats[:total_non_dfe_sign_ups]).to eq(1)
+    expect(stats[:total_candidate_count]).to eq(1)
     expect(stats[:candidates_with_submitted_forms]).to eq(1)
     expect(stats[:candidates_signed_in_but_not_entered_data]).to eq(0)
   end
+
+  it 'classifies application forms in the same way as ProcessState' do
```

Comment:
> How could the application form states unit test be made more performant, while still maintaining coverage? It's very slow at the moment but there are lots of nasty CASE...WHEN statements in the SQL query to cover

Here's an alternative way of testing this: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/commit/e47351495ac635cd8935969d908dbe19aa1449b2

It would look a lot like the process state spec: 

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/master/spec/services/process_state_spec.rb

It's much faster to use factories directly (`TestApplications.create_application` is optmised for realism, not speed). It also allows to individually test the states & counts - I think in the current test there's no test to make sure that we haven't accidentally swapped two state counts.

Reply:
Thanks – I've incorporated your approach in 98c97d8